### PR TITLE
Consolidate wake lock logic into StateFragment

### PR DESCRIPTION
**Pull request by phillipm**
Move a lot of wake lock dispatching out of CommCareActivity, consolidating it into StateFragment.

Also, rename and re-order some methods.

Probably easiest to review one commit at a time.

__Only review after https://github.com/dimagi/commcare-odk/pull/464 is merged__

### CODE REVIEW

File: app/src/org/commcare/android/framework/StateFragment.java [hunk 1]
```diff
@@ -10,95 +10,98 @@
...
 
-    private WakeLock wakelock;
-
-    public synchronized void wakelock(int lockLevel) {
-        if(wakelock != null) {
-            if(wakelock.isHeld()) {
-                wakelock.release();
+    private synchronized void acquireWakeLock() {
+        int lockLevel = boundActivity.getWakeLockLevel();
+        if (lockLevel != CommCareTask.DONT_WAKELOCK) {
```
Comment (ctsims) [inline app/src/org/commcare/android/framework/StateFragment.java]: Are we 100% sure this value can't intersect with the wakelock levels we request from the power manager?

**Comment by ctsims** [comment]
@phillipm Good to pull after answering my question

File: app/src/org/commcare/android/framework/StateFragment.java (hunk 1 above)
Comment (phillipm) [inline app/src/org/commcare/android/framework/StateFragment.java]: @ctsims I'm not, but I didn't change it in this PR. CommCareTask.DONT_WAKELOCK is a final set to -1

**Comment by phillipm** [comment]
@ctsims this PR depends on https://github.com/dimagi/commcare-odk/pull/464, could you take a look it and merge it first?

**Comment by phillipm** [comment]
@ctsims, @amsagoff merged 
https://github.com/dimagi/commcare-odk/pull/464 , so this ready to go again.

**Pull request merged**